src/components/basic/Badge.tsx

```diff
@@ -6,55 +6,109 @@
 
 import React, { useMemo, forwardRef, HTMLAttributes } from 'react';
 
-import styled from 'styled-components';
+import styled, { DefaultTheme } from 'styled-components';
 
+import { Icon } from './icon/Icon';
+import { getColor, useTheme } from '../../theme/theme-utils';
+import { AnyColor } from '../../types/utils';
 import { Tooltip } from '../display/Tooltip';
 
-const Comp = styled.span<{ isRead: boolean; isNumber: boolean }>`
-	display: inline-block;
-	min-width: 2em;
-	padding: ${({ isNumber, theme }): string =>
-		isNumber
-			? theme.sizes.padding.extrasmall
-			: `${theme.sizes.padding.extrasmall} ${theme.sizes.padding.small}`};
+type CompProps = {
+	$backgroundColor: AnyColor;
+	$color: AnyColor;
+	$padding: string;
+	$isBadgeCircle: boolean;
+};
+
+const Comp = styled.span<CompProps>`
```

Comment:
We should consider using a Container, which is a flex and let us set most of the styles directly as props